test/e2e/cni.go

```diff
@@ -0,0 +1,109 @@
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+)
+
+const (
+	calicoHelmChartRepoURL   string = "https://projectcalico.docs.tigera.io/charts"
+	calicoOperatorNamespace  string = "tigera-operator"
+	CalicoSystemNamespace    string = "calico-system"
+	CalicoAPIServerNamespace string = "calico-apiserver"
+	calicoHelmReleaseName    string = "projectcalico"
+	calicoHelmChartName      string = "tigera-operator"
+	ipv4                     string = "ipv4"
+	ipv6                     string = "ipv6"
+	dualStack                string = "dual-stack"
+)
+
+// InstallCalicoHelmChart installs the official calico helm chart
+// and validates that expected pods exist and are Ready.
+func InstallCalicoHelmChart(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput, cidrBlocks []string) {
+	specName := "calico-install"
+
+	By("Installing Calico CNI via helm")
+	values := getCalicoValues(cidrBlocks)
+	InstallHelmChart(ctx, input, calicoOperatorNamespace, calicoHelmChartRepoURL, calicoHelmChartName, calicoHelmReleaseName, values)
+	clusterProxy := input.ClusterProxy.GetWorkloadCluster(ctx, input.ConfigCluster.Namespace, input.ConfigCluster.ClusterName)
+	workloadClusterClient := clusterProxy.GetClient()
+
+	By("Waiting for Ready tigera-operator deployment pods")
+	for _, d := range []string{"tigera-operator"} {
+		waitInput := GetWaitForDeploymentsAvailableInput(ctx, clusterProxy, d, calicoOperatorNamespace, specName)
+		WaitForDeploymentsAvailable(ctx, waitInput, e2eConfig.GetIntervals(specName, "wait-deployment")...)
+	}
+	By("Waiting for Ready calico-system deployment pods")
+	for _, d := range []string{"calico-kube-controllers", "calico-typha"} {
+		waitInput := GetWaitForDeploymentsAvailableInput(ctx, clusterProxy, d, CalicoSystemNamespace, specName)
+		WaitForDeploymentsAvailable(ctx, waitInput, e2eConfig.GetIntervals(specName, "wait-deployment")...)
+	}
+	By("Waiting for Ready calico-apiserver deployment pods")
+	for _, d := range []string{"calico-apiserver"} {
+		waitInput := GetWaitForDeploymentsAvailableInput(ctx, clusterProxy, d, CalicoAPIServerNamespace, specName)
+		WaitForDeploymentsAvailable(ctx, waitInput, e2eConfig.GetIntervals(specName, "wait-deployment")...)
+	}
+	By("Waiting for Ready calico-node daemonset pods")
+	for _, ds := range []string{"calico-node"} {
+		WaitForDaemonset(ctx, input, workloadClusterClient, ds, CalicoSystemNamespace)
+	}
+}
+
+func getCalicoValues(cidrBlocks []string) *helmVals.Options {
+	var ipv6, ipv4 string
+	var values *helmVals.Options
+	for _, cidr := range cidrBlocks {
+		if k8snet.IsIPv6String(cidr) {
+			ipv6 = cidr
+		} else {
+			ipv4 = cidr
+		}
+	}
+	addonsPath := e2eConfig.GetVariable(capi_e2e.CNIPath)
+	if ipv6 != "" && ipv4 != "" {
+		// dual-stack
+		values = &helmVals.Options{
+			StringValues: []string{fmt.Sprintf("installation.calicoNetwork.ipPools[0].cidr=%s", ipv4), fmt.Sprintf("installation.calicoNetwork.ipPools.ipPools[1].cidr=%s", ipv6)},
```

Comment:
Do we plan on using the ipPool CIDRs specified in the different values.yaml files? Or is it just there as a default?

Reply:
The ipPool CIDRs in the values.yaml are there as default (they match the default values in cluster-template.yaml) so the file can be used outside of e2e tests, for example in the quickstart instructions